core-backend/sandbox/src/funcs.rs

```diff
@@ -455,13 +460,13 @@ where
     pub fn random(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
         sys_trace!(target: "syscall::gear", "random, args = {}", args_to_str(args));
 
-        let (subject_ptr, len, bn_random_ptr): (_, u32, _) = args.iter().read_3()?;
+        let (subject_ptr, bn_random_ptr): (_, _) = args.iter().read_2()?;
 
         ctx.run(|ctx| {
-            let mut subject = RuntimeBuffer::try_new_default(len as usize)?;
-            ctx.read_memory_into_buf(subject_ptr, subject.get_mut())?;
+            let mut subject =
```

Comment:
We don't need runtime buffer here for hash case

Reply:
Yes, checking constant size buffers is overwork.